### PR TITLE
more documentation for bselect

### PR DESCRIPTION
**Pull request by mlear**
Further documentation for BSelect and examples

### CODE REVIEW

File: src/components/BSelect/BSelect.vue [hunk 1]
```diff
@@ -293,3 +311,41 @@ export default {
...
+      :required=true
+      :selectOptions="[{id: 1, text: 'This'}, {id: 2, text: 'Is'}, {id: 3, text: 'An Example'}]"
+      id="example-id-3"
+      selectLabel="Pre Selected Example"
+    />
+  </div>
+</div>
+```
+
+</docs>
```
Comment (ghost-) [inline src/components/BSelect/BSelect.vue]: I think we were going to move the docs into a ComponentName.md (BSelect.md) file.  Do you want to just do that now?

File: src/components/BSelect/BSelect.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+```js
+
+<div>
+  <div style="width: 50%">
+    <b-select
+      :required=false
```
Comment (eleanormal) [inline src/components/BSelect/BSelect.md]: Can required be set by just having "required" as a property without a value?

File: src/components/BSelect/BSelect.vue [hunk 3]
```diff
@@ -61,25 +61,43 @@ export default {
...
   props: {
+    /**
+    * Component Id
+    */
     id: {
       type: String,
       required: true
     },
+    /**
+    * Array of objects to be displayed as options.
```
Comment (eleanormal) [inline src/components/BSelect/BSelect.vue]: We should define here what attributes need to be set on those objects

File: src/components/BSelect/BSelect.vue [hunk 4]
```diff
@@ -61,25 +61,43 @@ export default {
...
     },
+    /**
+    * Label of component.
+    */
     selectLabel: {
       type: String,
       required: true
     },
+    /**
+    * If included, this object will be pre-selected.
```
Comment (eleanormal) [inline src/components/BSelect/BSelect.vue]: It should be clear here that this needs to be the id of one of the selectOptions

File: src/components/BSelect/BSelect.vue [hunk 5]
```diff
@@ -61,25 +61,43 @@ export default {
...
     selectOptions: {
       type: Array
     },
+    /**
+    * Use to turn on validation and to indicate field is required.
+    */
     required: {
       type: Boolean,
       required: true
     },
```
Comment (ghost-) [inline src/components/BSelect/BSelect.vue]: Can this just be a boolean and not 'required'

For example ->

```js
required: {
type:Boolean
}
```

Do you want them to have to add a prop that is `required=true` or `required=false`

File: src/components/BSelect/BSelect.vue (hunk 3 above)
Comment (mlear) [reply src/components/BSelect/BSelect.vue]: I'll add this in my refactor branch

File: src/components/BSelect/BSelect.vue (hunk 4 above)
Comment (mlear) [reply src/components/BSelect/BSelect.vue]: I'll add this in my refactor branch

File: src/components/BSelect/BSelect.vue (hunk 5 above)
Comment (mlear) [reply src/components/BSelect/BSelect.vue]: I'll add this in my refactor branch

File: src/components/BSelect/BSelect.md (hunk 2 above)
Comment (mlear) [reply src/components/BSelect/BSelect.md]: I'll add this in my refactor branch

File: src/components/BSelect/BSelect.vue (hunk 4 above)
Comment (mlear) [reply src/components/BSelect/BSelect.vue]: its actually the entire object.

**Pull request merged**